src/ast/nodes/shared/Node.ts

```diff
@@ -222,10 +225,10 @@ export class NodeBase extends ExpressionEntity implements ExpressionNode {
 			if (value === null) continue;
 			if (Array.isArray(value)) {
 				for (const child of value) {
-					child?.include(context, includeChildrenRecursively);
+					child?.includePath(path, context, includeChildrenRecursively);
```

Comment:
I do not think this is a good default. We cannot know if including a certain path for the parent would include the same path for the child. I would vote for using `UNKNOWN_PATH` here to include all possible paths, and if nodes want something different, they should not call super.includePath but implement their own logic.